### PR TITLE
fix: external Interrupt priority problem (UART RX IRQ just worked once)

### PR DESCRIPTION
**Pull request by cathiele**
Working fix for #3

### CODE REVIEW

File: board/raspberrypi/raspberrypi-pico2/patches/linux/0002-patch-interrupt-controller.patch [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+ 
++#define CSR_MEICONTEXT 0xbe5
+ #define CSR_MEINEXT 0xbe4
+ #define CSR_MEIEA 0xbe0
+ 
+@@ -85,6 +86,7 @@
+ 	generic_handle_domain_irq(irqc->domain, hwirq);
+ 
+ 	chained_irq_exit(chip, desc);
++	csr_write(CSR_MEICONTEXT, 0x00);
```
Comment (Mr-Bossman) [inline board/raspberrypi/raspberrypi-pico2/patches/linux/0002-patch-interrupt-controller.patch]: what was the reason for this again. also super thanks for this!!!!

**Review by Mr-Bossman — APPROVED** [review]

**Comment by Mr-Bossman** [comment]
do you have access to the merge button? if so you can merge it when your ready

File: board/raspberrypi/raspberrypi-pico2/patches/linux/0002-patch-interrupt-controller.patch (hunk 1 above)
Comment (cathiele) [reply board/raspberrypi/raspberrypi-pico2/patches/linux/0002-patch-interrupt-controller.patch]: On entering the IRQ handler for external interrupts the hazard 3 sets the minimum enabled external interrupt priority to the priority of the currently handled interrupt + 1 in the ```MEICONTEXT``` CSR.

if no trap happens until ```mret``` this is restored automatically. But this does not work the way up from our interrupt handler back to the kernel riscv interrupt handling implementiation.

For now, setting the priority in ``MEICONTEXT`` to zero again would be fine, but we are ignoring the hardware priorities. Thats why in implemented the logic to properly handle the external interrupt priorities. 

Restoring it should happen somewhere "near" ```mret``` but that would be outside of our interrupt handler. 

For now it works as it is.

**Comment by cathiele** [comment]
> do you have access to the merge button? if so you can merge it when your ready

I don't have access to the merge button, but i am finished for now. Maybe you can merge it please?

**Pull request merged**